src/NewTools-FontChooser/StFontChooserPresenterTest.class.st

```diff
@@ -8,16 +8,53 @@ Class {
 	#package : 'NewTools-FontChooser'
 }
 
+{ #category : 'running' }
+StFontChooserPresenterTest >> newAllFamilyFontChooser: aFamilyName [
+
+	^ FontChooser newWithAllFamilies
+		font: (LogicalFont familyName: aFamilyName pointSize: 12);
+		yourself.	
+]
+
+{ #category : 'running' }
+StFontChooserPresenterTest >> newAllFamilyFontChooserFont: aLogicalFont [
+
+	^ FontChooser newWithAllFamilies
+		font: aLogicalFont;
+		yourself.	
+]
+
+{ #category : 'running' }
+StFontChooserPresenterTest >> newDefaultFamilyFontChooser: aFamilyName [
+
+	^ self newDefaultFamilyFontChooserFont: (LogicalFont familyName: aFamilyName pointSize: 12)
+]
+
+{ #category : 'running' }
+StFontChooserPresenterTest >> newDefaultFamilyFontChooserFont: aLogicalFont [
+
+	^ FontChooser newWithDefaultFamilies
+		font: aLogicalFont;
+		yourself.	
+]
+
+{ #category : 'tests' }
+StFontChooserPresenterTest >> openPresenterShouldntRaiseError [
+
+	self shouldnt: [ 
+			| window | 
+			window := presenter open.
+			window close ] raise: Error
```

Comment:
The goal of a test is to capture an error if any.
Then `shouldnt: raise: Error` is useless but the test is also useless. 
The method should just be named openPresenter.

Reply:
I don't understand. The presenter configured in the setUp fails to open if the fix in the PR is not applied. As stated in the issue:

```smalltalk
| fontChooser presenter |

fontChooser := FontChooser newWithAllFamilies
	title: 'Title';
	font: (LogicalFont
		familyName: 'Arial'
		pointSize: 11
		stretchValue: 5
		weightValue: 300
		slantValue: 0);
	onAcceptDo: [ : e | self halt ].
presenter := StFontChooserPresenter on: fontChooser.
presenter open.
```

See the following screenshot:
![Screenshot 2023-12-11 at 16 46 45](https://github.com/pharo-spec/NewTools/assets/4825959/eda516a5-f206-4741-93df-4ae1383b29a9)

The method is named openPresenterShouldntRaiseError since it is common code called from multiple tests:

```
StFontChooserPresenterTest>>#testOpenAllOnNonExistantFontFamily StFontChooserPresenterTest>>#testOpenDefaultOnUnloadedFontFamily StFontChooserPresenterTest>>#testOpenDefaultOnNonExistantFontFamily StFontChooserPresenterTest>>#testOpenAllOnUnloadedFontFamily
```

Reply:
What I mean is that a method test should never include:
```smalltalk
self shouldnt: [ ] raise: Error
```
It is the purpose of the test framework to capture all Errors and to make your test fail if any.
It would only make sense if you wanted to check for a specific error that should not happen.
You should rename `StFontChooserPresenterTest>>#openPresenterShouldntRaiseError` to `StFontChooserPresenterTest>>#openPresenter` with:
```smalltalk
openPresenter

	| window | 
	window := presenter open.
	window close 
```